services/src/main/java/org/exoplatform/officeonline/AbstractOfficeOnlineService.java

```diff
@@ -111,15 +111,20 @@ public AbstractOfficeOnlineService(SessionProviderService sessionProviders,
    * @param uuid the uuid
    * @param workspace the workspace
    * @return the node
-   * @throws RepositoryException the repository exception
    */
-  protected Node nodeByUUID(String uuid, String workspace) throws RepositoryException {
-    if (workspace == null) {
-      workspace = jcrService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName();
+  protected Node nodeByUUID(String uuid, String workspace) {
+    try {
+      if (workspace == null) {
+        workspace = jcrService.getCurrentRepository().getConfiguration().getDefaultWorkspaceName();
+      }
+      SessionProvider sp = sessionProviders.getSessionProvider(null);
+      Session userSession = sp.getSession(workspace, jcrService.getCurrentRepository());
+      return userSession.getNodeByUUID(uuid);
+    } catch (RepositoryException e) {
```

Comment:
ItemNotFoundException will tell that node not found by such UUID which supposed to be "client error", others RepositoryException may tell about real problems and better throw them out from the method to let be handled by the caller.

Reply:
Fixed: https://github.com/exo-addons/office-online/pull/4/commits/855e28198dcd8caf46d27be4e6d513e89c8ea7ed